crates/utils/src/utils/validation.rs

```diff
@@ -84,27 +84,37 @@ fn has_newline(name: &str) -> bool {
   name.contains('\n')
 }
 
-pub fn is_valid_actor_name(name: &str, actor_name_max_length: usize) -> LemmyResult<()> {
-  static VALID_ACTOR_NAME_REGEX_EN: LazyLock<Regex> =
-    LazyLock::new(|| Regex::new(r"^[a-zA-Z0-9_]{3,}$").expect("compile regex"));
-  static VALID_ACTOR_NAME_REGEX_AR: LazyLock<Regex> =
-    LazyLock::new(|| Regex::new(r"^[\p{Arabic}0-9_]{3,}$").expect("compile regex"));
-  static VALID_ACTOR_NAME_REGEX_RU: LazyLock<Regex> =
-    LazyLock::new(|| Regex::new(r"^[\p{Cyrillic}0-9_]{3,}$").expect("compile regex"));
+pub fn is_valid_username(name: &str, actor_name_max_length: usize) -> LemmyResult<()> {
+  // Only allow characters from a single alphabet per username. This avoids problems with lookalike
+  // characters like `o` which looks identical in Latin and Cyrillic, and can be used to imitate
+  // other users. Checks for additional alphabets can be added in the same way.
+  static VALID_USERNAME_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(r"^(?:[a-zA-Z0-9_]{3,}|[0-9_\p{Arabic}]{3,}|[0-9_\p{Cyrillic}]{3,})$")
+      .expect("compile regex")
+  });
 
-  let check = name.chars().count() <= actor_name_max_length && !has_newline(name);
+  is_valid_actor_name(name, actor_name_max_length, &VALID_USERNAME_REGEX)
+}
 
+pub fn is_valid_community_name(name: &str, actor_name_max_length: usize) -> LemmyResult<()> {
   // Only allow characters from a single alphabet per username. This avoids problems with lookalike
   // characters like `o` which looks identical in Latin and Cyrillic, and can be used to imitate
   // other users. Checks for additional alphabets can be added in the same way.
-  let lang_check = VALID_ACTOR_NAME_REGEX_EN.is_match(name)
-    || VALID_ACTOR_NAME_REGEX_AR.is_match(name)
-    || VALID_ACTOR_NAME_REGEX_RU.is_match(name);
+  static VALID_COMMUNITY_NAME_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(
+      r"^(?:[0-9_a-z]{3,}|[0-9_[\p{Arabic}]&&\P{Lu}&&\P{Lt}]{3,}|[0-9_[\p{Cyrillic}&&\P{Lu}&&\P{Lt}]]{3,})$",
```

Comment:
Omitting uppercase and titlecase characters.

Reply:
Why should uppercase be allowed for usernames but not for communities?

And same thing about the length check, I would also use `.is_lowercase()` in Rust.

Reply:
[Here](https://github.com/LemmyNet/lemmy-ui/pull/525) is the change for disallowing uppercase characters, but it was [reverted](https://github.com/LemmyNet/lemmy-ui/pull/541) a while later. For api tests we generate [random usernames and community names including uppercase characters](https://github.com/LemmyNet/lemmy/blob/main/api_tests/src/shared.ts#L887), so at least for the backend there should be no problem. So the backend works fine with uppercase, but maybe there would be problems in lemmy-ui. Anyway I cant find any reason why the rules should be different for username and community name, probably just an oversight.